proposals/0351-regex-builder.md

```diff
@@ -852,193 +1062,100 @@ extension Repeat {
   public init<W, C0, Component: RegexComponent>(
     _ component: Component,
     count: Int,
-    _ behavior: QuantificationBehavior = .eagerly
+    _ behavior: RepetitionBehavior? = nil
   )
   where Output == (Substring, C0),
-        Component.Output == (Substring, C0),
+        Component.RegexOutput == (Substring, C0),
         R.Bound == Int
   
   public init<W, C0, Component: RegexComponent>(
     count: Int,
-    _ behavior: QuantificationBehavior = .eagerly,
+    _ behavior: RepetitionBehavior? = nil,
     @RegexComponentBuilder _ component: () -> Component
   )
   where Output == (Substring, C0),
-        Component.Output == (Substring, C0),
+        Component.RegexOutput == (Substring, C0),
         R.Bound == Int
   
   public init<W, C0, Component: RegexComponent, RE: RangeExpression>(
     _ component: Component,
     _ expression: RE,
-    _ behavior: QuantificationBehavior = .eagerly
+    _ behavior: RepetitionBehavior? = nil
   )
   where Output == (Substring, C0?),
-        Component.Output == (W, C0),
+        Component.RegexOutput == (W, C0),
         R.Bound == Int
   
   public init<W, C0, Component: RegexComponent, RE: RangeExpression>(
     _ expression: RE,
-    _ behavior: QuantificationBehavior = .eagerly,
+    _ behavior: RepetitionBehavior? = nil,
     @RegexComponentBuilder _ component: () -> Component
   )
   where Output == (Substring, C0?),
-        Component.Output == (W, C0),
+        Component.RegexOutput == (W, C0),
         R.Bound == Int
   
   // ... `O(arity)` overloads
 }
 ```
 
-### Capture and reference
+</details>
 
-`Capture` and `TryCapture` produce a new `Regex` by inserting the captured pattern's whole match (`.0`) to the `.1` position of `Output`. When a transform closure is provided, the whole match of the captured content will be transformed to using the closure.
+#### Repetition behavior
 
-```swift
-public struct Capture<Output>: RegexComponent {
-  public var regex: Regex<Output> { get }
-}
-
-public struct TryCapture<Output>: RegexComponent {
-  public var regex: Regex<Output> { get }
-}
-```
+Repetition behavior defines how eager a repetition component should match the input. All repetition regex components have eager behavior by default.
```

Comment:
```suggestion
Repetition behavior defines how eager a repetition component should match the input. Behavior can be unspecified, in which case it will default to `eager` unless an option is provided to change the default (see [Unicode for String Processing](https://github.com/apple/swift-experimental-string-processing/blob/main/Documentation/Evolution/ProposalOverview.md#unicode-for-string-processing))
```